### ISSUE TITLE
update-mods fails on second mod for all instances.

### ISSUE THREAD

**Issue by Fendelgast** [FezVrasta/ark-server-tools#679]
Hello,

tl;dr when running arkmanager update update-mods the first mod downloads for all 3 instances, the second fails.

**Backstory.**

I recently split my server that was running 3 instances from 1 install directory, into 3 install directories. This was done because cronjobs that ran would update 1 install, but not restart the 3 instances that were running from it, causing 1 ark to restart, while 2 remained out of date. I took the opportunity to upgrade the server from Ubuntu 14.04 to 16.04.

**Install.**

Followed https://developer.valvesoftware.com/wiki/SteamCMD#Linux
created steam user, and installed arkmanager while running as "sudo -iu steam"
ran arkmanager install as "sudo -iu steam" also.

OS: Ubuntu 16.04
OS on SDA 1
Ark on SDA 2 (seperate raid array)

Steam Home dir:
$home/steam/

ArkManager Location:
$etc/arkmanager

Ark locations:
$hdd/ark/TC/
$hdd/ark/TI/
$hdd/ark/SE/

**Issue.**

When I run arkmanager update --update-mods the following occurs:
_$ arkmanager update --update-mods
No instances supplied for command update --update-mods
Do you wish to run this command for all instances?y
Running command 'update' for instance 'island'
2016-12-09 19:19:29: checking for update; PID: 3446
Downloading mod 554678442 ... -                                                                                                                                                                                                    Mod 554678442 downloaded
Downloading mod 708807240/usr/local/bin/arkmanager: line 466: home/steam/steamcmd.sh: No such file or directory
 ...
Mod 708807240 download failed
Mod 708807240 was not successfully downloaded
Your server is already up to date! The most recent version is .
Running command 'update' for instance 'center'
2016-12-09 19:19:36: checking for update; PID: 3596
Downloading mod 554678442 ... Mod 554678442 downloaded
Downloading mod 708807240 ... -/usr/local/bin/arkmanager: line 466: home/steam/steamcmd.sh: No such file or directory
Mod 708807240 download failed
Mod 708807240 was not successfully downloaded
Your server is already up to date! The most recent version is .
Running command 'update' for instance 'earth'
2016-12-09 19:19:42: checking for update; PID: 3743
Downloading mod 554678442 ... Mod 554678442 downloaded
Downloading mod 708807240 ... -/usr/local/bin/arkmanager: line 466: home/steam/steamcmd.sh: No such file or directory
Mod 708807240 download failed
Mod 708807240 was not successfully downloaded
Your server is already up to date! The most recent version is ._

**Troubleshooting performed:**
verified home/steam/steamcmd.sh exists, and is accessable
last ditch efforts 
chmod 7777 home/steam/*
chmod 7777 -R hdd/ark/*

Removed all mods from all servers, and re-installed via arkmanager installmod <modid>

**Comment by klightspeed** [comment]
Please ensure `steamcmdroot` is the absolute path to the steamcmd directory (i.e. `/home/steam` if you installed it there)

**Comment by Fendelgast** [comment]
Here's the top part of my arkmanager.cfg added / before home.

_arkstChannel="master"                                               # change it to a different branch to get non-stable versions
install_bindir="/usr/local/bin"
install_libexecdir="/usr/local/libexec/arkmanager"
install_datadir="/usr/local/share/arkmanager"

# config SteamCMD
steamcmdroot="/home/steam"                                                      # path of your steamcmd instance
steamcmdexec="steamcmd.sh"                                                      # name of steamcmd executable
steamcmd_user="steam"                                                          # name of the system user who own steamcmd folder
steamcmd_appinfocache="/hdd/ark/ARK-Logs/appinfo.vdf"      # cache of the appinfo command
steamcmd_workshoplog="/hdd/ark/ARK-Logs/workshop_log.txt" 
#steamlogin="anonymous"                                                         # Uncomment this to specify steam login instead of using anonymous login

# config environment
arkserverexec="ShooterGame/Binaries/Linux/ShooterGameServer"                  # name of ARK server executable
arkbackupdir="ARK-Backups"                                                    # path to backup directory
arkwarnminutes="15"                                                             # number of minutes to warn players when using update --warn
#arkautorestartfile="/hdd/steam/ark/ShooterGame/Saved/.autorestart"             # path to autorestart file
#arkautorestartfile='ShooterGame/Saved/.autorestart'
#arkAutoUpdateOnStart="true"                                                    # set this to true if you want to always update before startup
arkBackupPreUpdate="true"                                                       # set this to true if you want to perform a backup before updating
arkStagingDir="/hdd/ark/ARK-Staging"                                          # Uncomment to enable updates to be fully downloaded before restarting the server (reduces downtime while updating)

# Options to automatically remove old backups to keep backup size in check
# Each compressed backup is generally about 1-2MB in size.
arkMaxBackupSizeMB="500"                                                        # Set to automatically remove old backups when size exceeds this limit
#arkMaxBackupSizeGB="2"                                                         # Uncomment this and comment the above to specify the limit in whole GB_


**result is:**

$ arkmanager update --update-mods
No instances supplied for command update --update-mods
Do you wish to run this command for all instances?y
Running command 'update' for instance 'island'
2016-12-09 19:53:05: checking for update; PID: 14163
Downloading mod 554678442 ... Mod 554678442 was not successfully downloaded
Your server is already up to date! The most recent version is 1507914.
Running command 'update' for instance 'center'
2016-12-09 19:53:15: checking for update; PID: 14336
Downloading mod 554678442 ... Mod 554678442 was not successfully downloaded
Your server is already up to date! The most recent version is 1507914.
Running command 'update' for instance 'earth'
2016-12-09 19:53:23: checking for update; PID: 14486
Downloading mod 554678442 ... Mod 554678442 was not successfully downloaded
Your server is already up to date! The most recent version is 1507914.

**Comment by klightspeed** [comment]
What do the last few lines of `~steam/Steam/logs/workshop_log.txt` say?

**Comment by Fendelgast** [comment]
~steam/Steam/logs/ does not exist ? hmm.

I however just ran arkmanager reinstallmod <random modid> and had a permission denied for /home/steam/..../QW532T (temp) display. I adjusted permissions for the folder to 775, 

Guessing steam didn't install 100% or I had some odd permission issue specific to myself, seems to be working now.

So far:

$ arkmanager update --update-mods
No instances supplied for command update --update-mods
Do you wish to run this command for all instances?y
Running command 'update' for instance 'island'
2016-12-09 19:57:58: checking for update; PID: 16033
Downloading mod 708807240 ... Mod 708807240 downloaded
Downloading mod 718412068 ... Mod 718412068 downloaded
Downloading mod 719928795 ... Mod 719928795 downloaded
Downloading mod 722649005 ... Mod 722649005 downloaded
Downloading mod 731604991 ... Mod 731604991 downloaded
Downloading mod 747539326 ... Mod 747539326 downloaded
Downloading mod 748358288 ... -

**Comment by rob-bb** [comment]
I to am having problems updating modstoday, my workshop log has lots of this in it for different mods

```[2016-12-11 06:36:18] [AppID 346110] Loaded workshop items in "/home/steam/steamcmd/steamapps/workshop" (89 installed, 93 needed)
[2016-12-11 06:36:19] [AppID 346110] Starting Workshop download job
[2016-12-11 06:36:21] [AppID 346110] Download item 786896932 result : OK
[2016-12-11 06:36:21] [AppID 346110] Finished Workshop download job : No Error
[2016-12-11 06:36:21] [AppID 346110] Updating workshop item details failed with Failure```

This is for a range of mods, not just a single mod

**Comment by klightspeed** [comment]
@rob-bb that suggests `$steamcmdroot/steamapps/workshop/appworkshop_346110.acf` is corrupt or has bad permissions.  Try removing it then attempting to install the problematic mods again.
